src/SFML/Audio/SoundBuffer.cpp

```diff
@@ -68,15 +69,14 @@ SoundBuffer::SoundBuffer(const SoundBuffer& copy) : m_samples(copy.m_samples), m
 ////////////////////////////////////////////////////////////
 SoundBuffer::~SoundBuffer()
 {
-    // To prevent the iterator from becoming invalid, move the entire buffer to another
-    // container. Otherwise calling resetBuffer would result in detachSound being
-    // called which removes the sound from the internal list.
-    SoundList sounds;
-    sounds.swap(m_sounds);
-
-    // Detach the buffer from the sounds that use it (to avoid OpenAL errors)
-    for (Sound* soundPtr : sounds)
-        soundPtr->resetBuffer();
+    // Make sure no sound is attached to this buffer
+    if (!m_sounds.empty())
+    {
+        const char* plural = 2 <= m_sounds.size() ? "s" : "";
+        err() << "Failed to destruct sound buffer because it is used by " << m_sounds.size() << " sound" << plural
+              << std::endl;
```

Comment:
```suggestion
        err() << "Failed to destruct sound buffer because it is used by " << m_sounds.size() << " sound(s)" << plural
              << std::endl;
```
There's already a simpler way of expressing potential plurality.

Reply:
Good idea.